CorsixTH/Lua/humanoid_actions/die.lua

```diff
@@ -114,15 +114,20 @@ local action_die_tick_reaper; action_die_tick_reaper = permanent"action_die_tick
       hole_x, hole_y = humanoid.world.pathfinder:findIdleTile(spawn_scenario[2], spawn_scenario[3], 0)
 
       if hole_x and humanoid.world:canNonSideObjectBeSpawnedAt(hole_x, hole_y, "gates_to_hell", holes_orientation, 0, 0) then
-        if holes_orientation == "east" then
+        if holes_orientation == "south" then
           mirror_grim = 1
         end
         grim_use_tile_x = hole_x + spawn_scenario[4]
         grim_use_tile_y = hole_y + spawn_scenario[5]
         humanoid.hole_use_tile_x = hole_x + spawn_scenario[7]
         humanoid.hole_use_tile_y = hole_y + spawn_scenario[8]
-        -- tile can't be in a room
-        if humanoid.world:getRoom(humanoid.hole_use_tile_x, humanoid.hole_use_tile_y) then
+        -- tile can't be in a room and must be accessible by the patient
+        if not humanoid.world:getPathDistance(humanoid.tile_x, humanoid.tile_y, humanoid.hole_use_tile_x, humanoid.hole_use_tile_y)
+            or humanoid.world:getRoom(humanoid.hole_use_tile_x, humanoid.hole_use_tile_y) then
+          return false
+        end
+        -- ensure grim won't be in a a room
```

Comment:
Small typo here "a a"

Reply:
@mugmuggy missed this one ^?

Reply:
My bad, I also observed something I need to check related to lava hole. Completely unrelated to this change, so I might see if I can replicate that and include that with this.

Reply:
I wasn't able to replicate the error, maybe it was a different save than I was thinking, playing around with a few different ones when it occured, but the error was around persistence in player_hospital line 272. So I guess progress with this anyway.